databricks/koalas/series.py

```diff
@@ -5139,7 +5141,10 @@ def negative_check(s):
             return F.sum(F.log(negative_check(scol)))
 
         kser = self._cum(cumprod, skipna, part_cols)
-        return kser._with_new_scol(F.exp(kser.spark.column))
+        result = kser._with_new_scol(F.exp(kser.spark.column))
+        if isinstance(data_type, LongType):
```

Comment:
Shall we make the type checking the same as the other similar functions?

Reply:
Indeed, we should handle `IntegerType`, `ShortType`, and `ByteType` as well.

Reply:
Thanks, @ueshin !!